extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -118,6 +117,20 @@ export class AmpStoryQuiz extends AMP.BaseElement {
    * @private
    */
   attachContent_() {
+    const gridLayer = this.element.parentElement;
+
+    // Adjust grid-layer to identify quiz layers
+    gridLayer.classList.add('i-amphtml-has-quiz');
+
+    // Adjust grid-layer to identify CTAs and Page Attachments
+    if (gridLayer.parentElement.querySelector('amp-story-cta-layer')) {
+      gridLayer.classList.add('i-amphtml-has-CTA-layer');
+    }
+
+    if (gridLayer.parentElement.querySelector('amp-story-page-attachment')) {
+      gridLayer.classList.add('i-amphtml-has-page-attachment');
+    }
+
```

Comment:
I'm actually wondering if this block shouldn't be at the `amp-story-page` level.
@newmuis wdyt? This could potentially be used by other components in the future.

Additionally, it'd be great bonus if we didn't have any assumption that the quiz has to be a direct descendant of amp-story-grid-layer

Reply:
I modified this so I use the `closer` function, which should safely find the `amp-story-grid-layer`